java/code/src/com/redhat/rhn/frontend/xmlrpc/system/monitoring/SystemMonitoringHandler.java

```diff
@@ -0,0 +1,62 @@
+/**
+ *
+ * Copyright (c) 2021 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package com.redhat.rhn.frontend.xmlrpc.system.monitoring;
+
+import com.redhat.rhn.domain.dto.EndpointInfo;
+import com.redhat.rhn.domain.user.User;
+import com.redhat.rhn.frontend.xmlrpc.BaseHandler;
+import com.redhat.rhn.manager.formula.FormulaManager;
+
+import java.util.List;
+
+/**
+ * ServerMonitoringHandler
+ * @xmlrpc.namespace system.monitoring
+ * @xmlrpc.doc Provides methods to access information about managed systems, applications and formulas which can be
+ * relevant for Prometheus monitoring
+ */
+public class SystemMonitoringHandler extends BaseHandler {
+
+    private final FormulaManager formulaManager;
+
+    /**
+     * Instantiates a new system handler for system.monitoring namespace
+     * @param formulaManagerIn instance of formula manager object
+     */
+    public SystemMonitoringHandler(FormulaManager formulaManagerIn) {
+        this.formulaManager = formulaManagerIn;
+    }
+
+    /**
+     * Get the endpoint details for all Prometheus exporters installed on the systems whose IDs match
+     * with the passed systems IDs and all of the groups those systems are member of.
+     *
+     * @param loggedInUser The current user
+     * @param systemIDs The system IDs
+     * @return a list containing endpoint details for all Prometheus exporters on the passed system IDs.
+     *
+     * @xmlrpc.param #session_key()
+     * @xmlrpc.param #array_single("int", "systemID")
+     * @xmlrpc.returntype
+     *   #array_begin()
+     *     $EndpointInfo
+     *   #array_end()
+     */
+    public List<EndpointInfo> listEndpoints(User loggedInUser, List<Long> systemIDs) {
```

Comment:
Did you try this endpoint (eg. from a Python script)?

I was under the impression that in order to return a custom type one really had to add a custom `Serializer` class (see for example `com.redhat.rhn.frontend.xmlrpc.serializer.SystemOverviewSerializer`).

If that is wrong then I'll be happy to learn that plain beans are also completely OK to return!

Reply:
Good point. I've added custom serializer.